### DISCUSSION TITLE
Some async functions hang while running reach stdlib

### DISCUSSION THREAD

**Discussion by hmzdot** [reach-sh/reach-lang#929] [Q&A] [answered]
```js
const { loadStdlib } = require("@reach-sh/stdlib")
const { assert } = require("chai")

const reach = loadStdlib("ALGO")

describe("Simple Suite", function() {
    this.timeout(100000000)

    async function setup() {
        const startBal = reach.parseCurrency(100)
        console.log(startBal)

        const accs = await reach.newTestAccounts(2, startBal)
        console.log(accs)
        return accs
    }

    it("Test #1", function(done) {
        setup().then((accs) => { 
            assert.lengthOf(accs, 2, `expected accs len 2 got ${accs.length}`)
            done() 
        })
    })

    it("Test #2", async function() {
        try {
            const accs = await setup()
            console.log(accs)
        }
        catch(e) { console.log(e) }

        assert.lengthOf(accs, 2, `expected accs len 2 got ${accs.length}`)
    })
})
```

Both ''Test 1" and "Test 2" appear to hang when execution comes to the `newTestAccounts` part while running the contracts with `npx mocha`

**Comment by hmzdot** [comment]
This might be related to running stdlib on node environment because a similar thing happens when the code is ran without mocha and chai

**Reply by jeapostrophe** [reply]
Every test program and the tutorials all run with node and we obey all the normal JS/node patterns. I would investigate your node setup, your network setup, and if the JS package you're using is blocking or hiding activity. Running with debugging messages may show what is happening behind the scenes.

**Reply by Lucaster13** [reply]
I have gotten reach to run tests before but for some reason since I've gotten a new machine, reach no longer works.

My Setup:

Node 18.1.0
npm 8.8.0

I created a completely clean project and the package.json only imports reach

package.json:

```
{
    "name": "app",
    "version": "0.1.0",
    "private": true,
    "type": "module",
    "dependencies": {
        "@reach-sh/stdlib": "^0.1.10-rc.4"
    },
    "scripts": {
    },
    "devDependencies": {
    }
}
```

test.mjs:

```
import { loadStdlib } from "@reach-sh/stdlib";

const reach = loadStdlib("ALGO");

(async () => {
    console.log("here");
    const provider = await reach.getProvider();
    console.log(provider);
    const accs = await reach.newTestAccount(reach.parseCurrency(10));
    console.log(accs);
})();
```

I am trying to run `node test.mjs` and the program prints "here", but then never resolves the promises. I am sort of at a loss for how to debug this. Was hoping starting a fresh project might pinpoint the issue, but unfortunately this still hangs in limbo. 

Is your testing setup using a different node/npm version?

Thanks!

**Reply by hmzdot** [reply]
I think it is possible to replicate this by starting a new project with `npm init -y; npm install @reach-sh/stdlib` then running `node index.js` where index.js is

```js
const { loadStdlib } = require("@reach-sh/stdlib")
 
 const reach = loadStdlib("ALGO")
 
 const main = async () => {
     const startBal = reach.parseCurrency(100)
     console.log(startBal)
 
     await reach.wait("2")
 }
 
 main()
 .then(() => { process.exit(0) })
 .catch((e) => { console.error(e); process.exit(-1) })
```

**Reply by jeapostrophe** [reply]
I recommend running with `REACH_DEBUG=Y` and you'll be able to see what is being waited on

**Comment by hmzdot** [comment, marked as answer]
This was because most stdlib functions require devnet to be running in the background and as it will try to connect to the devnet it looks like halting.

Starting a development network using `./reach devnet` and then running tests works fine

**Reply by Lucaster13** [reply]
Amazing, it's working now, thanks for the help!

**Reply by michaelsermoneta** [reply]
I have the same problem 
![image](https://user-images.githubusercontent.com/1564473/189587057-522a2c76-8da3-4e7f-8952-0887ed0bf680.png), devnet is running, any advice? The code stop at const accCreator = await stdlib.newTestAccount(startingBalance);, the error is Error: connect ECONNREFUSED 127.0.0.1:8545\n at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1159:16)\n at TCPConnectWrap.callbackTrampoline (internal/async_hooks.js:130:17)

**Reply by jeapostrophe** [reply]
My guess is that the devnet is running in WSL2 but you're running the JS in VSCode in Windows, so localhost (127.0.0.1) is the Windows one, not the WSL2 one.

**Reply by michaelsermoneta** [reply]
thank you Jeapostrophe, I think you are right. My IDE is Intellij Ultimate on Windows, while devnet is running in WSL (Ubuntu). What is the solution? Find a way to run the devnet on Windows?

**Reply by jeapostrophe** [reply]
My advice is to do the opposite and try to run the program in WSL2, such as by running it with `reach run`

**Reply by michaelsermoneta** [reply]
Thank you. I already made it work with the "reach run" command, but I was trying to make the debugger work with the breakpoints, so i needed node js. I solved changing this line
const stdlib = loadStdlib();
to 
const stdlib = loadStdlib('ALGO');
I'm totally new to reach , so this could be the wrong solutions, I don't know.

**Reply by DanBurton** [reply]
`loadStdlib()` will look at your environment variables by default, and will use the input if specified. `loadStdlib('ALGO')` is a valid way to do it. You may, alternatively, want to consider using `REACH_CONNECTOR_MODE=ALGO` in whatever environment is used to run node.js.

**Reply by jeapostrophe** [reply]
Yea, it looks like you were trying to connect to an ETH devnet but you actually started an ALGO devnet. Your original post didn't say that you were trying to do ALGO, so I assumed the ETH error message was intentional
